pytorch/training/docker/2.0/py3/cu118/Dockerfile.gpu

```diff
@@ -533,6 +540,34 @@ RUN pip install --no-cache-dir -U ${SMD_MODEL_PARALLEL_URL}
 # Install SM Distributed DataParallel binary
 RUN SMDATAPARALLEL_PT=1 pip install --no-cache-dir ${SMD_DATA_PARALLEL_URL}
 
+# Install NCCL-tests (needed for Sagemaker provided health check)
+RUN git clone https://github.com/NVIDIA/nccl-tests.git ${HOME}/nccl-tests \
+    && cd ${HOME}/nccl-tests \
+    && git checkout \
+    && make MPI=1 MPI_HOME=/opt/amazon/openmpi/ CUDA_HOME=/opt/conda NCCL_HOME=/opt/nccl/lib \
+    && cp /root/nccl-tests/build/all_gather_perf /usr/bin \
+    && cp /root/nccl-tests/build/all_reduce_perf /usr/bin \
+    && cp /root/nccl-tests/build/alltoall_perf /usr/bin \
+    && cp /root/nccl-tests/build/broadcast_perf /usr/bin \
+    && cp /root/nccl-tests/build/gather_perf /usr/bin \
+    && cp /root/nccl-tests/build/hypercube_perf /usr/bin \
+    && cp /root/nccl-tests/build/reduce_perf /usr/bin \
+    && cp /root/nccl-tests/build/reduce_scatter_perf /usr/bin \
+    && cp /root/nccl-tests/build/scatter_perf /usr/bin \
+    && cp /root/nccl-tests/build/sendrecv_perf /usr/bin
+
+# Install Nvidia DCGM (needed for Sagemaker provided health check)
+RUN wget --content-disposition -P /tmp ${DCGM_URL}
+RUN ls -lt /tmp
+RUN dpkg -i /tmp/datacenter-gpu-manager_3.2.5_amd64.deb && rm -f /tmp/datacenter-gpu-manager_3.2.5_amd64.deb
```

Comment:
use `&&` to make all DCGM commands together, same for EFA checker, also make sure all not needed files are removed once packages are installed

Reply:
@jeet4320 Can we make sure we provide all the comments all in once? We are working on this PR for over 3 weeks

Reply:
Changed